presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/NoopMetastoreCache.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.metastore;
+
+import com.facebook.presto.common.predicate.Domain;
+import com.facebook.presto.hive.ForCachingHiveMetastore;
+import com.facebook.presto.hive.HiveTableHandle;
+import com.facebook.presto.spi.constraints.TableConstraint;
+import com.facebook.presto.spi.security.PrestoPrincipal;
+import com.facebook.presto.spi.security.RoleGrant;
+
+import javax.inject.Inject;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Noop Metastore cache which does not cache any data
+ * All the calls are routed to the underlying delegate
+ */
+public class NoopMetastoreCache
```

Comment:
Not a fan of name Noop, its doing something, its calling the delegate.

Reply:
Even better might be to avoid using the CachingHiveMetastore entirely when we aren't using the caching. Just use the delegate directly.

Reply:
removed this since we can turn off by avoiding the CachingPath if needed.